src/app/util/ember-compatibility-functions.cpp

```diff
@@ -479,6 +479,22 @@ CHIP_ERROR WriteSingleClusterData(ClusterInfo & aClusterInfo, TLV::TLVReader & a
     attributePathParams.mFieldId    = aClusterInfo.mFieldId;
     attributePathParams.mFlags.Set(AttributePathParams::Flags::kFieldIdValid);
 
+    // TODO: Refactor WriteSingleClusterData and all dependent functions to take ConcreteAttributePath instead of ClusterInfo
+    // as the input argument.
+    AttributeAccessInterface * attrOverride = findAttributeAccessOverride(aClusterInfo.mEndpointId, aClusterInfo.mClusterId);
+    if (attrOverride != nullptr)
+    {
+        bool dataWrite;
+        ConcreteAttributePath path(aClusterInfo.mEndpointId, aClusterInfo.mClusterId, aClusterInfo.mFieldId);
+
+        ReturnErrorOnFailure(attrOverride->Write(path, aReader, &dataWrite));
+
+        if (dataWrite)
+        {
+            return apWriteHandler->AddStatus(attributePathParams, Protocols::InteractionModel::Status::Success);
+        }
+    }
+
     auto imCode = WriteSingleClusterDataInternal(aClusterInfo, aReader, apWriteHandler);
```

Comment:
Shouldn't this be protected with `if (!dataWrite)`?

Reply:
no need, since the function is returned if dataWrite is true